crml/eth-bridge/src/lib.rs

```diff
@@ -212,10 +222,19 @@ decl_module! {
 				ProcessedTxBuckets::remove_prefix(expired_bucket_index, None);
 
 				// TODO: better estimate
-				50_000_000_u64
-			} else {
-				Zero::zero()
+				weight += 50_000_000 as Weight;
 			}
+
+			if !Self::bridge_paused() {
+				weight = weight.saturating_add(DbWeight::get().reads(1 as Weight));
```

Comment:
`reads * delayed_events_per_block()` + `writes * delayed_events_per_block()` for the `::remove` part.
```suggestion
				weight = weight.saturating_add(DbWeight::get().reads(1 as Weight ));
```